internal/integration/integration_test.go

```diff
@@ -53,316 +69,323 @@ func TestIntegration(t *testing.T) {
-	}
+		keys := util.GenerateExposureKeys(3, -1, false)
 
-	// Assert there are 3 exposures in the database
-	{
-		exposures, err := getExposures(db, criteria)
-		if err != nil {
-			t.Fatal(err)
+		// Publish 3 keys
+		payload := &verifyapi.Publish{
+			Keys:              keys,
+			HealthAuthorityID: "com.example.app",
 		}
-		if got, want := len(exposures), 3; got != want {
-			t.Fatalf("expected %v to be %v", got, want)
+		jwtCfg.ExposureKeys = keys
+		jwtCfg.JWTWarp = tc.JWTWrap
+		verification, salt := testutil.IssueJWT(t, jwtCfg)
+		payload.VerificationPayload = verification
+		payload.HMACKey = salt
+		var revisionToken string
+		if resp, err := client.PublishKeys(payload); err != nil {
+			if !strings.Contains(err.Error(), tc.Error) {
+				t.Fatalf("expected error: %v, got: %v", tc.Error, err)
+			}
+			continue
+		} else {
+			if resp.RevisionToken == "" {
+				t.Fatal("empty revision token")
+			}
+			revisionToken = resp.RevisionToken
+			t.Logf("response: %+v", resp)
 		}
-	}
 
-	// Get the exported exposures
-	var exported *export.TemporaryExposureKeyExport
-	Eventually(t, 30, func() error {
-		// Trigger an export
-		if err := client.ExportBatches(); err != nil {
-			return err
+		// Assert there are 3 exposures in the database
+		{
+			exposures, err := getExposures(db, criteria)
+			if err != nil {
+				t.Fatal(err)
+			}
+			if got, want := len(exposures), 3; got != want {
+				t.Fatalf("expected %v to be %v", got, want)
+			}
 		}
 
-		// Start export workers
-		if err := client.StartExportWorkers(); err != nil {
-			return err
-		}
+		// Get the exported exposures
+		var exported *export.TemporaryExposureKeyExport
+		Eventually(t, 30, func() error {
+			// Trigger an export
+			if err := client.ExportBatches(); err != nil {
+				return err
+			}
 
-		// Attempt to get the index
-		index, err := env.Blobstore().GetObject(ctx, ExportDir, IndexFilePath())
-		if err != nil {
-			if errors.Is(err, storage.ErrNotFound) {
-				return retry.RetryableError(fmt.Errorf("Can not find index file: %v", err))
+			// Start export workers
+			if err := client.StartExportWorkers(); err != nil {
+				return err
 			}
-			return err
-		}
 
-		// Find the latest file in the index
-		lines := strings.Split(string(index), "\n")
-		latest := ""
-		for _, entry := range lines {
-			if strings.HasSuffix(entry, "zip") {
-				if latest == "" {
-					latest = entry
-				} else {
-					if entry > latest {
+			// Attempt to get the index
+			index, err := env.Blobstore().GetObject(ctx, ExportDir, IndexFilePath())
+			if err != nil {
+				if errors.Is(err, storage.ErrNotFound) {
+					return retry.RetryableError(fmt.Errorf("Can not find index file: %v", err))
+				}
+				return err
+			}
+
+			// Find the latest file in the index
+			lines := strings.Split(string(index), "\n")
+			latest := ""
+			for _, entry := range lines {
+				if strings.HasSuffix(entry, "zip") {
+					if latest == "" {
 						latest = entry
+					} else {
+						if entry > latest {
```

Comment:
I know this was carried over from prior, but if "" is the smallest lexicographically. Is there really a need for the  `latest == ""` condition? I think `if entry > latest` covers all cases

Reply:
Done